go/timedlock/rw_lock.go

```diff
@@ -0,0 +1,192 @@
+package timedlock
```

Comment:
Lets see if we can get #3255 merged with this change and have you rebase this on devel instead